### DISCUSSION TITLE
Failed updating repository: permission denied

### DISCUSSION THREAD

**Discussion by TheZoker** [semaphoreui/semaphore#1267] [Q&A]
When I try to run a task, I get the following error:
```
8:12:49 PM
Task 6 added to queue
8:12:51 PM
Preparing: 6
8:12:51 PM
Prepare TaskRunner with template: LXC Alpine Update
8:12:51 PM
Cloning Repository <git-url>
8:12:51 PM
Failed updating repository: open /tmp/semaphore/access_key_335449536: permission denied
```

I'm running semaphore via docker compose. When I enter the docker container, there is no file or folder within the `/tmp/semaphore/` path. 

Can someone help me fix this? That would be amazing!

**Comment by jrglynn2** [comment]
I don't know docker, but the `/tmp/semaphore/access_key_335449536` file is a temporary file that Semaphore creates when running a task. It is a representation of the Key File you are using that is configured in Semaphore.

From a pure linux perspective, the error is saying that Semaphore doesn't have access to write anything to the `/tmp/` directory - perhaps changing the permissions on that upper level folder will fix the issue?

**Comment by ansibleguy** [comment]
In the default docker-setup I would assume `/tmp/semaphore` would be writable.
Could it be you have any docker-volumes mapped/configured in your docker-compose file?
The docker-compose config you are using would be helpful.
